core/domain/topic_services.py

```diff
@@ -694,10 +694,65 @@ def get_topic_rights_from_model(topic_rights_model):
 
     return topic_domain.TopicRights(
         topic_rights_model.id,
-        topic_rights_model.manager_ids
+        topic_rights_model.manager_ids,
+        topic_rights_model.topic_is_published
     )
 
 
+def publish_topic(topic_id, committer_id):
+    """Marks the given topic as published.
+
+    Args:
+        topic_id: str. The id of the given topic.
+        committer_id: str. ID of the committer.
+
+    Raises:
+        Exception. The given topic does not exist.
+        Exception. The user does not have enought rights to publish the topic.
+    """
+    topic_rights = get_topic_rights(topic_id, strict=False)
+    if topic_rights is None:
+        raise Exception('The given topic does not exist')
+    user_actions_info = user_services.UserActionsInfo(committer_id)
+    if not check_can_edit_topic(user_actions_info, topic_rights):
+        raise Exception(
+            'The user does not have enought rights to publish the topic.')
+
+    topic_rights.topic_is_published = True
+    commit_cmds = [{
```

Comment:
We should be using TopicRightsChange objects for these, right? Ditto below.

Reply:
Yeah, created it.